toolchain/sem_ir/ids.h

```diff
@@ -225,6 +229,10 @@ constexpr NameId NameId::SelfType = NameId(NameId::InvalidIndex - 2);
 constexpr NameId NameId::ReturnSlot = NameId(NameId::InvalidIndex - 3);
 constexpr NameId NameId::PackageNamespace = NameId(NameId::InvalidIndex - 4);
 constexpr NameId NameId::Base = NameId(NameId::InvalidIndex - 5);
+constexpr int NameId::SpecialValueCount = 5;
+// Enforce the link between SpecialValueCount and the last special value.
+static_assert(NameId::SpecialValueCount ==
+              NameId::InvalidIndex - NameId::Base.index);
```

Comment:
Given that the uses of this all add one, maybe it'd be simpler to include `InvalidIndex` in the count of special values?

Reply:
Echoing the discussion around #3575 and invalid, I would feel more comfortable distinguishing between a valid special value and invalid.

Reply:
Works for me. I think something like `LowestValidValue` could also work; using `id.index - LowestValidValue` to convert an ID to an array index seems like it'd be reasonably readable and clear.